File: server/php/src/Application/Services/PaymentLinkService.php
<?php

namespace OnlinePayments\ExampleApp\Application\Services;

use Exception;
use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\RequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\ResponseDto;
use OnlinePayments\ExampleApp\Application\Interfaces\SDKClients\PaymentLinkClientInterface;
use OnlinePayments\ExampleApp\Application\Interfaces\Services\PaymentLinkServiceInterface;

readonly class PaymentLinkService implements PaymentLinkServiceInterface
{
    public function __construct(
        private PaymentLinkClientInterface $paymentLinkClient,
    ) {}

    /**
     * @throws Exception
     */
    public function createPaymentLink(RequestDto $requestDto): ResponseDto
    {
        $responseDto = $this->paymentLinkClient->createPaymentLink($requestDto);

        $responseDto->amount = $requestDto->amount;
        $responseDto->currency = $requestDto->currency;

        return $responseDto;
    }
}

File: server/python/app/application/services/payment_link_service.py
from app.application.dtos.payment_link.request_dto import CreatePaymentLinkRequestDto
from app.application.dtos.payment_link.response_dto import CreatePaymentLinkResponseDto
from app.application.interfaces.sdk_clients.payment_link_client_interface import PaymentLinkClientInterface
from app.application.interfaces.services.payment_link_service_interface import PaymentLinkServiceInterface

class PaymentLinkService(PaymentLinkServiceInterface):

    def __init__(self, payment_link_client: PaymentLinkClientInterface):
        self._client = payment_link_client

    def create_payment_link(self, request_dto: CreatePaymentLinkRequestDto) -> CreatePaymentLinkResponseDto:
        response_dto = self._client.create_payment_link(request_dto)

        response_dto.amount = request_dto.amount
        response_dto.currency = request_dto.currency

        return response_dto